Improved the rift storage

Lauriichan opened the pull request:
This PR introduces a more performance friendly storage system for the rift inventory
It also allows to have more than 256 rifts (the old implemention used getByte("Rift") to get the id)
All rifts have a UUID instead of an numeric id and their own file making it easier to get rid of them

This PR also adds a random item generator block for debugging purposes if needed I can strip it again
To test stuff using this mod I added PipeZ, Spark, SolarGeneration and EdivadLib for testing, I commented for the time being

Lauriichan committed:
Add debug tools

Lauriichan committed:
Add translations

Lauriichan committed:
Restructure RiftStorage

Lauriichan committed:
Fully implemented more performant rift solution

Lauriichan committed:
Fixed debug item tag

Lauriichan committed:
Update DebugItemBlockEntity.java

Lauriichan committed:
Added migration to rift storage (to load old rifts)

Lauriichan committed:
Comment debugging mod dependencies

gigaherz reviewed (changes requested):
It seems your changes also include changing the formatting style of the code, making reviewing the changes almost impossible due to the unnecessary style changes all over the place. And frankly, I have my style preferences which I would rather keep.

Lauriichan commented:
> It seems your changes also include changing the formatting style of the code, making reviewing the changes almost impossible due to the unnecessary style changes all over the place. And frankly, I have my style preferences which I would rather keep.

That's mostly my auto formatter yeah, I didn't edit all files so it mostly only includes all files I edited
If you tell me more about the style you would like to see I can work on that

gigaherz commented:
> That's mostly my auto formatter yeah, I didn't edit all files so it mostly only includes all files I edited
> If you tell me more about the style you would like to see I can work on that

Main issue is the brace location, which I prefer in its own line.
But I'd also ask that your PR should not modify any code that you didn't change, and that any new code should be consistent with the code around it. Even if that requires disabling parts of the auto-formatter.

> EDIT: If you use eclipse's auto formatter I can also just import your settings if you could export them for me

I do not, I use Intellij IDEA, with mostly the default settings except for the things I changed once upon a time to match my preferences.

Lauriichan commented:
> But I'd also ask that your PR should not modify any code that you didn't change, and that any new code should be consistent with the code around it. Even if that requires disabling parts of the auto-formatter.

Yeah as I said no problem for me

Lauriichan committed:
Fix style

Lauriichan commented:
> I do not, I use Intellij IDEA, with mostly the default settings except for the things I changed once upon a time to match my preferences.

If the style isn't fine like this you could send me your IntelliJ IDEA settings and then I'll format it with that one once

gigaherz commented:
You'll notice that you still have A WHOLE LOT of differences. I didn't notice it before (because they are invisible), but your settings also replaced all my indentations by tabs, I prefer spaces.

[Default__1_.xml.txt](https://github.com/gigaherz/Ender-Rift/files/9365320/Default__1_.xml.txt)

**But let me be more direct with this**: I do not want you to find an autoformatter preset that is "close" to what I have, because my IDE settings don't do things like split long lines automatically, and I don't run the auto-formatting.

Your new code should use my choices with regards to brace locations and indentation type **and leave every other line in the file as it was before**. If you didn't modify a line of code, it should not exist in the code diff.

Lauriichan commented:
> If you didn't modify a line of code, it should not exist in the code diff.

Well easier said than done if I'm being honest after making my changes but I'll try my best to fix it

gigaherz commented:
> Well easier said than done if I'm being honest after making my changes but I'll try my best to fix it

What I would suggest you do:
1. reformat your code to 4 spaces and braces in their own lines.
2. In your favorite git client, use the option to "show diff between" / "show changes from" or whatever you client has, and choose the commit you used as a basis for the PR (the latest commit in the main repository, presumably). And go change by change, and apply the old version of the lines you didn't modify yourself.

gigaherz commented:
(sorry if anything I say comes through as rude or specific, it's not my intention, but I am halfway to a headache/migraine)

Lauriichan commented:
> (sorry if anything I say comes through as rude or specific, it's not my intention, but I am halfway to a headache/migraine)

Don't worry x)
I understand that someone just comming and changing your code style can be very annoying and it was not my intention to hurt anyone when adding this feature ^^
As I said I'll fix it x)

Lauriichan committed:
Update style

Lauriichan commented:
Okay now it should be fine ^^

Lauriichan committed:
Hopefully last style fix

Lauriichan committed:
Last style fixes

gigaherz commented:
I have started reviewing the code, but I am also in the middle of making supper so I will continue later.

Lauriichan commented:
Okay, take your time ^^
Theres no need to hurry or anything especially because its your mod

Lauriichan commented on src/main/java/dev/gigaherz/enderrift/rift/RiftStructure.java:
```diff
@@ -182,12 +183,14 @@ private static void buildStructure(Level world, BlockPos pos, ItemStack itemStac
-        else
-        {
-            rift.assemble(RiftStorage.get(world).getNextRiftId());
+            UUID id = tagCompound.getUUID("RiftId");
```
This might need migration as well as the Rift Core Item could still contain an old rift id

Lauriichan replied:
I'll add that later

Lauriichan replied:
[Used verifyTagAfterLoad](https://github.com/gigaherz/Ender-Rift/pull/56/commits/9872c64fd9350429e87f0b93aeea34bf7763d8c5#diff-4aea2f72b433984cdd51355dbbc472b0349cc5a1f90eaee263fa89fb0847c9d9R42) to migrate

Lauriichan committed:
Add migration to rift item

Try fix debug item block model (didn't work)
Small changes to the debug random item generator

Lauriichan committed:
Fix weird rightclick behaviour

Lauriichan committed:
Fix QuickMove behaviour as well

Lauriichan commented:
Hey there just wanted to check how things are going since a week already passed

gigaherz commented:
Oof sorry my ability to focus has gone to shit. I'll try to remember over the weekend.

Lauriichan commented:
Hey there I'm guessing you forgot about it again or had no time so I wanted to remind you of this issue again :)

gigaherz commented:
I uh.... haven't forgotten, just haven't found a combination of remembering, having time, and having energy. Sorry >_<

Lauriichan commented:
Well yeah that's understandable as well no worries ^^

Lauriichan commented:
Hey there its about 3 months now wanted to remind you of the PR because of that if you might have already forgotten about it ^^ If you still didn't have the time that's fine but I still wanted to remind you

gigaherz commented:
Oh, heck. I did forget this time. My situation hasn't really changed since September -- I have basically not been doing anything coding-related at all outside my day job.
I had some concerns with some of the changes, which I still haven't figured out.
But I'll merge now, and then figure things out locally. Expect some of your changes to get tweaked before I release this to the public. :P

Lauriichan commented:
> Oh, heck. I did forget this time. My situation hasn't really changed since September -- I have basically not been doing anything coding-related at all outside my day job. I had some concerns with some of the changes, which I still haven't figured out. But I'll merge now, and then figure things out locally. Expect some of your changes to get tweaked before I release this to the public. :P

Feel free to change whatever you like :)
I just wanted to help fix the lag issues and implemented it in a way that it should be better than before.
There were some things that I didn't understand of your systems so I tried to minimize the amount of changes to existing code as I didn't want to mess things up.

Also if you have any questions related to the code feel free to contact me per mail or if you want we could also message on discord
